projects/igniteui-angular/src/lib/chips/chip.spec.ts

```diff
@@ -144,4 +185,167 @@ describe('IgxChip', () => {
+
+        expect(firstChipIconName).toContain('drag_indicator');
+    });
+
+    it('should set chips suffix correctly', () => {
+        const fix = TestBed.createComponent(TestChipsLabelAndSuffixComponent);
+        fix.detectChanges();
+
+        const chipArea = fix.debugElement.queryAll(By.directive(IgxChipsAreaComponent));
+        const chipElements = chipArea[0].queryAll(By.directive(IgxChipComponent));
+        const firstChipSuffix = chipElements[0].queryAll(By.directive(IgxSuffixDirective));
+        const firstChipSuffixText = firstChipSuffix[0].nativeElement.innerHTML;
+
+        expect(firstChipSuffixText).toEqual('suf');
+    });
+
+    it('should set chips suffix connector correctly', () => {
+        const fix = TestBed.createComponent(TestChipComponent);
+        fix.detectChanges();
+
+        const chipArea = fix.debugElement.queryAll(By.directive(IgxChipsAreaComponent));
+        const chipElements = chipArea[0].queryAll(By.directive(IgxChipComponent));
+        const firstChipSuffixConnector = chipElements[0].queryAll(By.directive(IgxConnectorDirective));
+        const firstChipSuffixConnectorIconName = firstChipSuffixConnector[0].nativeElement.textContent;
+
+        expect(firstChipSuffixConnectorIconName).toContain('forward');
+    });
+
+    it('should make chip comfortable when density is not set', () => {
+        const fix = TestBed.createComponent(TestChipComponent);
+        fix.detectChanges();
+
+        const components = fix.debugElement.queryAll(By.directive(IgxChipComponent));
+        const firstComponent = components[0];
+
+        expect(firstComponent.componentInstance.displayDensity).toEqual(DisplayDensity.comfortable);
+    });
+
+    it('should make chip comfortable when density is set to comfortable', () => {
+        const fix = TestBed.createComponent(TestChipComponent);
+        fix.detectChanges();
+
+        const components = fix.debugElement.queryAll(By.directive(IgxChipComponent));
+        const secondComponent = components[1];
+
+        expect(secondComponent.componentInstance.displayDensity).toEqual(DisplayDensity.comfortable);
+    });
+
+    it('should make chip compact when density is set to compat', () => {
+        const fix = TestBed.createComponent(TestChipComponent);
+        fix.detectChanges();
+
+        const components = fix.debugElement.queryAll(By.directive(IgxChipComponent));
+        const thirdComponent = components[2];
+
+        expect(thirdComponent.componentInstance.displayDensity).toEqual(DisplayDensity.compact);
+    });
+
+    it('should make chip cosy when density is set to cosy', () => {
+        const fix = TestBed.createComponent(TestChipComponent);
+        fix.detectChanges();
+
+        const components = fix.debugElement.queryAll(By.directive(IgxChipComponent));
+        const fourthComponent = components[3];
+
+        expect(fourthComponent.componentInstance.displayDensity).toEqual(DisplayDensity.cosy);
```

Comment:
@MartinKamenov Add another expect and check whether the correct CSS is applied. Right now you're verifying the model which is not enough.